typescript/cli/src/commands/options.ts

```diff
@@ -1,10 +1,96 @@
-// A set of common options
-import { Options } from 'yargs';
+/* TODO: Create common/options.ts, common/args.ts, agent/options.ts, etc. */
+import { ArgumentsCamelCase, Options } from 'yargs';
 
-export const keyCommandOption: Options = {
+type CommandOptions = {
+  chains: Options;
+};
+export type AgentCommandOptions = CommandOptions & {
+  origin: Options;
+  targets: Options;
+  config: Options;
+};
+export type CoreCommandOptions = CommandOptions & {
+  targets: Options;
+  artifacts: Options;
+  ism: Options;
+  hook: Options;
+  out: Options;
+  key: Options;
+  yes: Options;
+  'dry-run': Options;
+};
+export type WarpCommandOptions = CommandOptions & {
+  config: Options;
+  core: Options;
+  out: Options;
+  key: Options;
+  yes: Options;
+};
+
+interface CommandArgs extends ArgumentsCamelCase {
+  chains: string;
+}
+export interface AgentCommandArgs extends CommandArgs {
+  origin: string;
+  targets: string;
+  config: string;
+}
+export interface CoreCommandArgs extends CommandArgs {
+  targets: string;
+  artifacts: string;
+  ism: string;
+  hook: string;
+  out: string;
+  key: string;
+  yes: boolean;
+  'dry-run': boolean;
+}
+export type WarpCommandArgs = CommandArgs & {
+  config: string;
+  core: string;
+  out: string;
+  key: string;
+  yes: boolean;
+};
```

Comment:
is this all unused?

Reply:
Thought I might use this in next iteration, but decided to remove. Removed now